chrome/content/zotero/xpcom/proxy.js

```diff
@@ -69,6 +69,8 @@ Zotero.Proxies = new function() {
 		Zotero.Proxies.lastIPCheck = 0;
 		Zotero.Proxies.lastIPs = "";
 		Zotero.Proxies.disabledByDomain = false;
+
+		Zotero.Proxies.showRedirectNotification = Zotero.Prefs.get("proxies.showRedirectNotification");
```

Comment:
You should either update this when the pref changes — which you can do with a handler in ``Zotero.Prefs`` in zotero.js — or just check the pref value directly below. Otherwise this won't change until Zotero is restarted.

(It's possible this is all premature optimization and the latter would be fine, but I understand that the existing code does this. Not sure if anyone has tested repeated pref lookup vs. property access speed.)